### PR TITLE
Tracker retention duration parameter [DPP-191]

### PR DESCRIPTION
**Pull request by kamil-da**
**Motivation**
It has been established that there is a huge excess of command completion queries against the index database which is caused by the so called command trackers. They come alive whenever someone is using the CommandService on behalf of a particular client/party combination. They stay alive for the next 24h.
Modifying the tracker retention period could improve the performance of a ledger.

CHANGELOG_BEGIN
- New --tracker-retention-period parameter for kvutils CLI to be able to customize for how long the command service will keep an active command tracker for a given party.

CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-integration-api/src/main/scala/platform/configuration/CommandConfiguration.scala [hunk 1]
```diff
@@ -41,6 +41,6 @@ object CommandConfiguration {
       maxParallelSubmissions = 512,
       maxCommandsInFlight = 256,
       limitMaxCommandsInFlight = true,
-      retentionPeriod = 24.hours,
+      retentionPeriod = 5.minutes,
```
Comment (SamirTalwar-DA) [inline ledger/participant-integration-api/src/main/scala/platform/configuration/CommandConfiguration.scala:44]: This needs to go into the changelog.

File: ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala [hunk 2]
```diff
@@ -176,6 +182,15 @@ object Config {
           config.withTlsConfig(c => c.copy(enableCertRevocationChecking = checksEnabled))
         )
 
+      opt[Duration]("tracker-retention-period")
+        .optional()
+        .action((value, config) =>
+          config.copy(trackerRetentionPeriod = FiniteDuration(value.getSeconds, TimeUnit.SECONDS))
+        )
+        .text(
+          "For how long the command service will keep an active command tracker for a given party. A longer retention period allows to not instantiate a new tracker for a party that seldom acts. A shorter retention period allows to quickly remove unused trackers. Default is 5 minutes."
```
Comment (SamirTalwar-DA) [inline ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala:191]:
```suggestion
          s"For how long the command service will keep an active command tracker for a given party. A longer retention period allows to not instantiate a new tracker for a party that seldom acts. A shorter retention period allows to quickly remove unused trackers. Default is $DefaultTrackerRetentionPeriod."
```

File: ledger/participant-integration-api/src/main/scala/platform/configuration/CommandConfiguration.scala (hunk 1 above)
Comment (SamirTalwar-DA) [inline ledger/participant-integration-api/src/main/scala/platform/configuration/CommandConfiguration.scala:44]: Perhaps pull it out into a constant here, and use it in `kvutils/app/Config.scala`.

**Review by SamirTalwar-DA — COMMENTED** [review]
Are there any potential ramifications in doing this?

File: ledger/participant-integration-api/src/main/scala/platform/configuration/CommandConfiguration.scala (hunk 1 above)
Comment (kamil-da) [reply ledger/participant-integration-api/src/main/scala/platform/configuration/CommandConfiguration.scala:44]: Good point!

File: ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala (hunk 2 above)
Comment (kamil-da) [reply ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala]: Fixed.

File: ledger/participant-integration-api/src/main/scala/platform/configuration/CommandConfiguration.scala (hunk 1 above)
Comment (kamil-da) [reply ledger/participant-integration-api/src/main/scala/platform/configuration/CommandConfiguration.scala]: Fixed.

**Comment by kamil-da** [comment]
> Are there any potential ramifications in doing this?

@mziolekda - could you provide some information on this?

File: ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala [hunk 3]
```diff
@@ -176,6 +182,15 @@ object Config {
           config.withTlsConfig(c => c.copy(enableCertRevocationChecking = checksEnabled))
         )
 
+      opt[Duration]("tracker-retention-period")
+        .optional()
+        .action((value, config) =>
+          config.copy(trackerRetentionPeriod = FiniteDuration(value.getSeconds, TimeUnit.SECONDS))
+        )
+        .text(
+          s"For how long the command service will keep an active command tracker for a given party. A longer retention period allows to not instantiate a new tracker for a party that seldom acts. A shorter retention period allows to quickly remove unused trackers. Default is $DefaultTrackerRetentionPeriod."
```
Comment (mziolekda) [inline ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala:191]:
```suggestion
          s"How long will the command service keep an active command tracker for a given party. A longer period cuts down on the tracker instantiation cost for a party that seldom acts. A shorter period causes a quick removal of unused trackers. Default is $DefaultTrackerRetentionPeriod."
```

**Review by mziolekda — CHANGES_REQUESTED** [review]
A small change to the command line help.

**Review by mziolekda — APPROVED** [review]

**Pull request merged**